modules/darwinosl/darwinosl-source.m

```diff
@@ -0,0 +1,559 @@
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "msg_size_max");
+  stats_register_aggregator_maximum(self->options.super_source_options->super.stats_level, &sc_key,
+                                    &self->max_message_size);
+
+  stats_cluster_single_key_legacy_set_with_name(&sc_key,
+                                                self->options.super_source_options->super.stats_source | SCS_SOURCE,
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "msg_size_avg");
+  stats_register_aggregator_average(self->options.super_source_options->super.stats_level, &sc_key,
+                                    &self->average_messages_size);
+
+  stats_cluster_single_key_legacy_set_with_name(&sc_key,
+                                                self->options.super_source_options->super.stats_source | SCS_SOURCE,
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "eps");
+
+  stats_register_aggregator_cps(self->options.super_source_options->super.stats_level, &sc_key, &sc_key_eps_input,
+                                SC_TYPE_SINGLE_VALUE,
+                                &self->CPS);
+
+  stats_aggregator_unlock();
+}
+
+static void
+_unregister_aggregated_stats(DarwinOSLogSourceDriver *self)
+{
+  stats_aggregator_lock();
+
+  stats_unregister_aggregator_maximum(&self->max_message_size);
+  stats_unregister_aggregator_average(&self->average_messages_size);
+  stats_unregister_aggregator_cps(&self->CPS);
+
+  stats_aggregator_unlock();
+}
+
+static void
+_check_restored_postion(DarwinOSLogSourceDriver *self, OSLogEntry *nextLogEntry)
+{
+  NSString *msgString = [self->osLogSource stringFromDarwinOSLogEntry:nextLogEntry];
+
+  if (g_str_hash(msgString.UTF8String) == self->log_source_position.last_msg_hash)
+    {
+      msg_debug("darwinosl: Bookmark is restored fine",
+                evt_tag_str("bookmark", [OSLogSource.RFC3339DateFormatter stringFromDateWithMicroseconds:
+                                                                          nextLogEntry.date].UTF8String));
+    }
+  else
+    {
+      if (self->log_source_position.last_msg_hash)
+        msg_info("darwinosl: Could not restore last bookmark (filter might be changed or max_bookmark_distance took effect?)",
+                 evt_tag_str("bookmark", [OSLogSource.RFC3339DateFormatter stringFromDateWithMicroseconds:
+                                                                           [NSDate dateWithTimeIntervalSince1970:self->log_source_position.log_position]].UTF8String),
+                 evt_tag_str("new start position", [OSLogSource.RFC3339DateFormatter stringFromDateWithMicroseconds:
```

Comment:
please don't use spaces in the tag name. it needs to be parseable relatively easily and spaces break that format.

Reply:
just for the record, i just searched for a single type of tag (evt_tag_long),  and found a bunch of tag names with spaces, anyway, renamed them